### ISSUE TITLE
Are there any plans to implement a ZigBee stack on top of the esp-ieee802154 crate?

### ISSUE THREAD

**Issue by PortalRising** [esp-rs/esp-hal#1689]
Are there any plans to implement a ZigBee stack on top of the esp-ieee802154 crate, perhaps porting this repository,[https://github.com/zigpy/zigpy](zigpy), to rust?

**Comment by jessebraham** [comment]
Until there is an open-source ZigBee stack I can't imagine any of us will have the time for something as complex like this.

**Comment by bjoernQ** [comment]
There is some effort: https://github.com/blueluna/psila

Another alternative would be to use or port https://github.com/niclash/zboss

**Comment by thebino** [comment]
I took over the existing [zigbee crate](https://crates.io/crates/zigbee) and started to re-implement the stack from scratch based on the latest [ZigBee specification 22 1.0](https://csa-iot.org/wp-content/uploads/2022/01/docs-05-3474-22-0csg-zigbee-specification-1.pdf)

If there are any volunteers, we could speed up the development.

**Comment by giomf** [comment]
@thebino did you already made some progress? Im willing to help somehow.

**Comment by thebino** [comment]
@giomf indeed, you can find it here: [github.com/zigbee-rs/zigbee-rs](https://github.com/zigbee-rs/zigbee-rs)

I also added parts of the [ZigBee Cluster library](https://github.com/zigbee-rs/zigbee-cluster-library) for testing the send part, but that's open if we'll maintain it in the future.

**Comment by MabezDev** [comment]
We're looking forward to https://github.com/zigbee-rs/zigbee-rs flourishing, however this isn't something we can really track here in esp-hal. I'm closing this issue, but watching `zigbee-rs` closely!

**Comment by thebino** [comment]
@bjoernQ @jessebraham @MabezDev @PortalRising 
🎉 zigbee-rs 0.1.0 just got release
7 crates, `no_std`, `async`, `edition 2024`, `dual MIT/Apache-2.0`
